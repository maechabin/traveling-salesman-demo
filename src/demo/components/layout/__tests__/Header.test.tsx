import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('<DemoDescription />', () => {
  it('snapshot', () => {
    // setup
    const wrapper = shallow(<Header />);

    // snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
