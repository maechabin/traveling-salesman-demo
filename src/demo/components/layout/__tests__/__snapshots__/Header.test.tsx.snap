// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemoDescription /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={
            Object {
              "fontSize": "16px",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          サラリーマン巡回問題 ~最短経路を探せ！~
        </h1>,
        <p
          style={
            Object {
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "right": "16px",
              "top": 0,
            }
          }
        >
          <a
            href="https://github.com/maechabin/traveling-salesman-demo"
            rel="noopener noreferrer"
            style={
              Object {
                "color": "#fff",
              }
            }
            target="_blank"
          >
            GitHub
          </a>
        </p>,
      ],
      "style": Object {
        "backgroundColor": "#7fbc39",
        "color": "#fff",
        "lineHeight": "40px",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "サラリーマン巡回問題 ~最短経路を探せ！~",
          "style": Object {
            "fontSize": "16px",
            "margin": 0,
            "padding": 0,
          },
        },
        "ref": null,
        "rendered": "サラリーマン巡回問題 ~最短経路を探せ！~",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://github.com/maechabin/traveling-salesman-demo"
            rel="noopener noreferrer"
            style={
              Object {
                "color": "#fff",
              }
            }
            target="_blank"
          >
            GitHub
          </a>,
          "style": Object {
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "right": "16px",
            "top": 0,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GitHub",
            "href": "https://github.com/maechabin/traveling-salesman-demo",
            "rel": "noopener noreferrer",
            "style": Object {
              "color": "#fff",
            },
            "target": "_blank",
          },
          "ref": null,
          "rendered": "GitHub",
          "type": "a",
        },
        "type": "p",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
              Object {
                "fontSize": "16px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            サラリーマン巡回問題 ~最短経路を探せ！~
          </h1>,
          <p
            style={
              Object {
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "right": "16px",
                "top": 0,
              }
            }
          >
            <a
              href="https://github.com/maechabin/traveling-salesman-demo"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "#fff",
                }
              }
              target="_blank"
            >
              GitHub
            </a>
          </p>,
        ],
        "style": Object {
          "backgroundColor": "#7fbc39",
          "color": "#fff",
          "lineHeight": "40px",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "サラリーマン巡回問題 ~最短経路を探せ！~",
            "style": Object {
              "fontSize": "16px",
              "margin": 0,
              "padding": 0,
            },
          },
          "ref": null,
          "rendered": "サラリーマン巡回問題 ~最短経路を探せ！~",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://github.com/maechabin/traveling-salesman-demo"
              rel="noopener noreferrer"
              style={
                Object {
                  "color": "#fff",
                }
              }
              target="_blank"
            >
              GitHub
            </a>,
            "style": Object {
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "right": "16px",
              "top": 0,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "GitHub",
              "href": "https://github.com/maechabin/traveling-salesman-demo",
              "rel": "noopener noreferrer",
              "style": Object {
                "color": "#fff",
              },
              "target": "_blank",
            },
            "ref": null,
            "rendered": "GitHub",
            "type": "a",
          },
          "type": "p",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
